json/src/main/scala/fs2/data/json/internal/JsonTokenParser.scala

```diff
@@ -210,88 +212,95 @@ private class JsonTokenParser[F[_], T](s: Stream[F, T])(implicit F: RaiseThrowab
         case NumberState.Invalid =>
           if (NumberState.isFinal(state)) {
             T.appendMarked(context, acc)
-            chunkAcc += Token.NumberValue(acc.result())
+            chunkAcc.numberValue(acc.result())
             Pull.done
           } else
-            emitChunk(chunkAcc) >> Pull.raiseError[F](new JsonException(s"invalid number character '$c'"))
+            emitChunk() >> Pull.raiseError[F](new JsonException(s"invalid number character '$c'"))
         case state =>
           T.advance(context)
           number_(state, acc)
       }
     }
   }
 
-  private final def keyword_(expected: String, eidx: Int, elen: Int, token: Token): Pull[F, Token, Unit] = {
+  private final def keyword_(expected: String,
+                             eidx: Int,
+                             elen: Int,
+                             accumulate: () => ChunkAccumulator[Res]): Pull[F, Res, Unit] = {
     if (T.needsPull(context)) {
-      emitChunk(chunkAcc) >> T.pullNext(context).flatMap {
+      emitChunk() >> T.pullNext(context).flatMap {
         case Some(context) =>
           this.context = context
-          chunkAcc.clear()
-          keyword_(expected, eidx, elen, token)
+          chunkAcc.flush()
+          keyword_(expected, eidx, elen, accumulate)
         case None => Pull.raiseError[F](new JsonException("unexpected end of input"))
       }
     } else {
       val c = T.current(context)
       if (c == expected.charAt(eidx)) {
         if (eidx == elen - 1) {
           T.advance(context)
-          chunkAcc += token
+          accumulate()
```

Comment:
Can you quickly add a comment on what's happening here? I would expect a `() => ChunkAccumulator[Res]` with discarded result to be a no-op…reading the call sites below I suspect this emits the keyword-equivalent value via the facade and we don't need the result as we don't pass it on, but maybe we can make this clearer, like calling it `emitKeywordValue`?

Reply:
I am not a big fan of this part so far, but this is the best I came up with for now. Renaming will help, indeed.